src/app/plugins/dog.plugin.ts

```diff
@@ -1,66 +1,134 @@
+import { MessageEmbed } from 'discord.js';
 import Constants from '../../common/constants';
 import { Plugin } from '../../common/plugin';
-import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { ChannelType, IContainer, IHttpResponse, IMessage, Maybe } from '../../common/types';
 
 export class DogPlugin extends Plugin {
   public name: string = 'Dog Plugin';
   public description: string = 'Generates pictures of doggos.';
-  public usage: string = 'dog <breed>';
+  public usage: string = 'dog <breed (Optional)> | dog breed | dog subBreed | dog random';
   public pluginAlias = ['dogs', 'doggo'];
   public permission: ChannelType = ChannelType.Public;
   public pluginChannelName: string = Constants.Channels.Public.Pets;
 
-  private _API_URL: string = 'https://api.woofbot.io/v1/';
+  private _API_URL: string = 'https://dog.ceo/api/';
+
+  private _allBreeds: string[] = [];
   private _breeds: string[] = [];
+  private _subBreeds: _DogSubBreed[] = [];
+
+  private _breedEmbed: Maybe<MessageEmbed>;
+  private _subBreedEmbed: Maybe<MessageEmbed>;
 
   constructor(public container: IContainer) {
     super();
     this.container.httpService
-      .get(`${this._API_URL}breeds`)
+      .get(`${this._API_URL}breeds/list/all`)
       .then((response: IHttpResponse) => {
-        const breeds: string[] = response.data.response.breeds;
-        breeds.forEach((breed) => {
-          this._breeds.push(breed.toLowerCase());
+        const breedData = response.data.message;
+        this._breeds = this._allBreeds = Object.keys(breedData);
+
+        // The json is annoyingly {<breed>: [<subBreeds>]} apposed to {breed: <breed>, subBreed: [<subBreeds>]}} so this gets that
+        this._breeds.forEach((breed: string) => {
```

Comment:
maybe can we use a reduce or map here so that we aren't altering `_subBreeds` or `_allBreeds` in a side-effecty way

Reply:
Not quite sure what you're wanting, but changed the method I handle it so it isn't as, "oh yeah, and change these while you're at it".